evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -228,9 +252,16 @@ def predict_in_sample(self, X, y, X_train, y_train, objective=None):
         X, y = self._drop_time_index(X, y)
         X_train, y_train = self._drop_time_index(X_train, y_train)
         target = infer_feature_types(y)
-        features = self.transform_all_but_final(X, target, X_train, y_train)
+        features = self.transform_all_but_final(
+            X,
+            target,
+            X_train,
+            y_train,
+            calculating_residuals=calculating_residuals,
+        )
         predictions = self._estimator_predict(features)
-        predictions.index = y.index
+        if not calculating_residuals:
```

Comment:
So, what I've found and correct me if I'm wrong @eccabay , when we do a truly in-sample prediction the prediction size is not guaranteed to be the same size as the target values array...because of the lags I think.

Reply:
I think I need to go back and do a little more experimentation here. You're right @chukarsten that the lags and dropping null rows means that the prediction size won't be the same size as the target values array. However, that's not the case if we've precomputed features with the DFS transformer, I had to add the index alignment @jeremyliweishih [commented on](https://github.com/alteryx/evalml/pull/4127/files#r1160846874) to get this to work in the product. I'll do some experimenting to see if I can consolidate the two cases, and get something that makes a bit more sense.

Reply:
I revisited this, it looks like a simpler way to keep this safe is to just limit setting the indices to be equal for when `len(predictions) == len(y)`, it removes the need for the dfs check as well. I've updated accordingly but can always change it again if desired!